api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxMlQueries.java

```diff
@@ -857,96 +861,70 @@ public static String getPatientsTracedWithVisitCard(
-            + "                   INNER JOIN encounter e "
-            + "                   ON e.patient_id = p.patient_id "
-            + "                   INNER JOIN obs o "
-            + "                   ON o.encounter_id = e.encounter_id "
-            + "               WHERE  p.voided = 0 "
-            + "                   AND e.voided = 0 "
-            + "                   AND o.voided = 0 "
-            + "                   AND e.encounter_type IN (${adultoSeguimentoEncounterType},${ARVPediatriaSeguimentoEncounterType}) "
-            + "                   AND e.encounter_datetime >= :onOrAfter "
-            + "                   AND e.encounter_datetime <= :onOrBefore "
-            + "                   AND e.location_id = :location "
-            + "               GROUP  BY p.patient_id ) q1 "
-            + "               left join obs o2 on o2.encounter_id=q1.encounter_id and "
-            + "                   o2.concept_id = ${returnVisitDateConcept} and o2.voided=0 "
-            + "               where  o2.obs_id  is null) ficha "
-            + "           INNER JOIN ( "
-            + "               SELECT q2.patient_id "
-            + "               from ( "
-            + "                   SELECT p.patient_id, "
-            + "                       Max(e.encounter_datetime) as max_enc_datetime, Max(e.encounter_id) AS encounter_id "
-            + "                   FROM patient p "
-            + "                       INNER JOIN encounter e "
-            + "                       ON e.patient_id = p.patient_id "
-            + "                       INNER JOIN obs o "
-            + "                       ON o.encounter_id = e.encounter_id "
-            + "                   WHERE  p.voided = 0 "
-            + "                       AND e.voided = 0 "
-            + "                       AND o.voided = 0 "
-            + "                       AND e.encounter_type IN (${aRVPharmaciaEncounterType}) "
-            + "                       AND e.encounter_datetime >= :onOrAfter "
-            + "                       And e.encounter_datetime <= :onOrBefore "
-            + "                       AND e.location_id = :location "
-            + "                   GROUP  BY p.patient_id  "
-            + "               )q2 "
-            + "               left join obs o1 on o1.encounter_id=q2.encounter_id and "
-            + "                       o1.concept_id = ${returnVisitDateForArvDrugConcept} and o1.voided=0 "
-            + "               where  o1.obs_id is null "
-            + "           ) fila ON ficha.patient_id=fila.patient_id ) filaficha on filaficha.patient_id=pm.patient_id "
-            + "       )ps "
-            + "       GROUP  BY ps.patient_id";
+        "SELECT pat.patient_id "
+            + "FROM   patient pat "
+            + "WHERE  pat.voided=0 "
+            + "    AND    pat.patient_id NOT IN( "
+            + "        SELECT fn.patient_id "
```

Comment:
This already exists in txcurr

Reply:
just used the same implementation! Please check